content/docs/2.12/concepts/scaling-deployments.md

```diff
@@ -201,6 +201,80 @@ Starting from Kubernetes v1.18 the autoscaling API allows scaling behavior to be
 
 **Assumptions:** KEDA must be running on Kubernetes cluster v1.18+, in order to be able to benefit from this setting.
 
+---
+
+```yaml
+advanced:
+  complexScalingLogic:                                    # Optional. Section to specify complex scaling logic
+    target: {target-value-to-scale-on}                    # Mandatory. New target if metrics are anyhow composed together
+    formula: {formula-for-fetched-metrics}                # Mandatory. Formula for calculation
+```
+
+**`complexScalingLogic:`**
+
+The `complexScalingLogic` is optional. If defined, both `target` and `formula` are mandatory. Using this structure creates `composite-scaler-metric` for the HPA that will replace all requests for external metrics and handle then internally. With `complexScalingLogic` all external triggers' names **must** be defined, for potentially being used in the `formula` itself.
+
+**`complexScalingLogic.target`**
+
+`target` defines new target value to scale on for the composed metric. All external metrics must be of the same type in order for ScaledObject to be successfully validated.
+
+**`complexScalingLogic.formula`**
+
+  `formula` composes external metrics together and allows them to be modified/manipulated with. It accepts mathematical/conditional statements using [this external project](https://github.com/antonmedv/expr). If `fallback` is applied, the `formula` will NOT be applied. Complete language definition of `expr` package can be found [here](https://expr.medv.io/docs/Language-Definition). Formula must return a single value (not boolean).
+
+**Example 1**
```

Comment:
I'm thinking out loud - Everything above here is a nice overview, but all the rest below should go in a dedicated section similar to https://deploy-preview-1198--keda.netlify.app/docs/2.12/concepts/scaling-deployments/#pause-autoscaling.

This is because the samples are mixed with the conceptual doc which makes the whole section a bit message/unstructured. I'd do a brief introduction here, add a dedicated section down below and just point to it here. (or move everything in a dedicated section).

The current document for this section is already _way too long_

Thoughts @zroubalik / @JorTurFer?

Reply:
Agree!